src/devices/cpu/i386/i386.cpp

```diff
@@ -3944,9 +3959,77 @@ void i386_device::device_reset()
 
 	m_CPL = 0;
 
+	m_auto_clear_RF = true;
+
 	CHANGE_PC(m_eip);
 }
 
+inline void i386_device::dri_changed()
+{
+	int dr;
+	if(!(m_dr[7] & 0xff)) return;
+	for(dr = 0; dr < 4; dr++)
+	{
+		if(m_dr_breakpoints[dr]) m_dr_breakpoints[dr]->remove();
```

Comment:
Since you're removing all taps each time, you can use only one handler for all of them and call the remove once just before the loop.  It will be slightly faster

Reply:
There's already no perceivable speed difference as it is.